src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala

```diff
@@ -89,7 +89,7 @@ trait WorkerMain[S] {
                 System.err.println(s"Uncaught exception in Future while proccessing WorkRequest $requestId:")
                 e.printStackTrace(System.err)
               }
-            }
+            }(scala.concurrent.ExecutionContext.global)
```

Comment:
so you have any info about the differences between these two execution contexts? You mention "IO tasks" being handled by this default context. Would it make sense to use a cached/unbounded context? Or would that cause memory issues by the amount of threads we could end up with?

Reply:
I think that this is a pretty lightweight task. It's just writing to stdout so the worker can communicate with the bazel server, but it's doing so synchronized. I figure we might as well free up the part of the worker that's actually doing work to do that work and let the synchronized IO part happen in another thread pool.